craft_parts/plugins/ant_plugin.py

```diff
@@ -142,6 +145,15 @@ def get_build_packages(self) -> Set[str]:
     @override
     def get_build_environment(self) -> Dict[str, str]:
         """Return a dictionary with the environment to use in the build step."""
+        # Getting ant to use a proxy requires a little work; the JRE doesn't
+        # help as much as it should.  (java.net.useSystemProxies=true ought
+        # to do the trick, but it relies on desktop configuration rather
+        # than using the standard environment variables.)
```

Comment:
Thanks for this comment! My first question here was going to be whether we can put this upstream in the general Java plugin. (And perhaps at some point we can provide a lot of this logic in the java plugin, such as retrieving the proxy options, but it's probably a good idea to let it simmer here for a bit even if we do want to do that.)